File: mp2i/files/C5/puissance.ml
let rec puissance x n =
  if n=0 then 1 else x * puissance x (n-1)

let rec puissance_acc x n acc =
  if n=0 then acc else puissance_acc x (n-1) (x*acc) 

let rec puissance_rapide x n =
  if n=0 then 1 else
    if n mod 2 = 0 then 
      (puissance_rapide x (n/2) * puissance_rapide x (n/2)) 
else x * (puissance_rapide x (n/2) * puissance_rapide x (n/2))

File: site/itc/files/C16/puissance.py
# Calcule x puissance n, avec n>0
def puissance(x,n):
    p = 1
    for i in range(1,n):
        p = p * n
    return p

def puissance_rapide(x,n):
    if n==0:
        return 1
    p = puissance_rapide(x,n//2)
    if n%2==0:
        return p*p
    else:
        return p*p*x
        
    
